src/App.tsx

```diff
@@ -1,39 +1,57 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './Components/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
+type State = {
+  clockName: string,
+  isClockVisible: boolean,
+};
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+export class App extends Component<{}, State> {
+  state = {
+    clockName: 'Clock-0',
+    isClockVisible: true,
+  };
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.contextMenuHandler);
+    document.addEventListener('click', this.clickHandler);
+
+    window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('contextmenu', this.contextMenuHandler);
+    document.removeEventListener('click', this.clickHandler);
+  }
+
+  contextMenuHandler = (e: MouseEvent) => {
```

Comment:
let's look through the checklist once again

Reply:
changed method names from `clickHandler && contextMenuHandler` to `handleClick && handleContextMenu`